### PR TITLE
EOS-14345: Use setup.yaml to update BE LOG size in LDR R1

### PR DESCRIPTION
**Pull request by imvenkip**
Problem: We need to update BE log size during update cluster in
maintenance mode if MOTR_M0D_BELOG_SIZE change in /etc/sysconfig/motr

Solution: "/opt/seagate/cortx/motr/conf/setup.yaml" will trigger during
update cluster in maintenance mode. Added "be_log_resize" in setup.yaml
to trigger "m0provision be_log_resize". It will execute be_resize.sh to
modify BE log size if any change in /etc/sysconfig/motr.

Signed-off--by: Venkateswarlu Payidimarry <venkateswarlu.payidimarry@seagate.com>

### CODE REVIEW

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+BE_REPLY="be_recovery_run"
+BE_RESIZE="be_log_resize"
+
+local_md_volume=""
+remote_md_volume=""
+new_be_log_size=""
+current_be_log_size=""
+local_node=""
+remote_node=""
+loca_ios_fid=""
```
Comment (imvenkip) [inline scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:21]: s/loca_ios_fid/local_ios_fid

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	fi
+	be_resize_log "Completed unmount_md_device"
+}
+
+check_cluster_status
+get_be_log_size
+mount_md_device
+get_ios_fid
+verify_be_log_size
+be_log_reply
```
Comment (madhavemuri) [inline scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:191]: be_log_replay would be the right name for it.

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+	elif [[ $(cat /var/lib/hare/node-name) == "srvnode-2" ]]; then
+		local_node="srvnode-2"
+		remote_node="srvnode-1"
+		local_md_volume=$(grep right-volume /opt/seagate/cortx/ha/conf/build-ees-ha-args.yaml |\
+				    awk '{ print $2 }')
+		remote_md_volume=$(grep left-volume /opt/seagate/cortx/ha/conf/build-ees-ha-args.yaml |\
+				    awk '{ print $2 }')
+	else
+		be_resize_hard_exit "ERROR: node-name is not avaialble"
```
Comment (madhavemuri) [inline scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:85]: This is a valid case for LDR R2, needs to be handled, add a todo here.

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	fi
+
+	if ! run_cmd_on_local_node "mount $local_md_volume $MD_DIR"; then
```
Comment (madhavemuri) [inline scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:88]: There are two cases here, if log resize is run through HA module, need to run on
both the nodes and when provisioner invokes update to will run on each node independently. So need to differentiate these two cases.

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    echo "[`date`]: $@" 2>&1 | tee -a ${LOG_FILE}
+}
+
+be_resize_soft_exit() { be_resize_log "$@" ; exit 0; }
+
+be_resize_hard_exit() { be_resize_log "$@" ; exit 1; }
+
+# Verifying cluster status
+check_cluster_status() {
+	if  [[ $((hctl status) 2>&1 | grep "Cluster is not running") ]]; then
```
Comment (madhavemuri) [inline scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:34]: Check for m0d's too, which should not be running.

Comment (madhavemuri) [reply scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:34]: Also for LDR R1, if mount point exists, return from here.

Comment (imvenkip) [reply scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:34]: Handled two conditions.

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh (hunk 3 above)
Comment (imvenkip) [reply scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:85]: Done.

File: scripts/install/opt/seagate/cortx/motr/config/be_resize.sh (hunk 4 above)
Comment (imvenkip) [reply scripts/install/opt/seagate/cortx/motr/config/be_resize.sh:88]: be_resize.sh will execute in local node.  Modified code to support VM, LDR1 and LDR2 .

**Comment by imvenkip** [comment]
BE log resize design has changed.  Created new ticket EOS-14600.

**Pull request closed without merging**